setup.py

```diff
@@ -45,26 +44,30 @@ def run_tests(self):
 
 class BuildDocsCommand(setuptools.command.build_py.build_py):
     apidoc_command = (
-        'sphinx-apidoc', '-f', '-o', 'docs', '--no-toc', 'coalib'
+        'sphinx-apidoc', '-f', '-o', 'docs',
+        '--no-toc',
+        'coalib'
     )
-    doc_command = ('make', '-C', 'docs', 'html', 'SPHINXOPTS=-W')
+    make_command = ('make', '-C', 'docs', 'html', 'SPHINXOPTS=-W')
 
     def run(self):
-        errOne = call(self.apidoc_command)
-        errTwo = call(self.doc_command)
-        sys.exit(errOne or errTwo)
+        err_no = call(self.apidoc_command)
+        if not err_no:
+            err_no = call(self.make_command)
```

Comment:
Even though the new version is better performance wise (no idea how much) I find the fist one a lot easier to read.

Reply:
This is the one used by coala-bears: https://github.com/coala/coala-bears/blob/5ca3eb7/setup.py#L34

Here is us fixing the same bug twice .. ;-)

https://github.com/coala/coala/commit/069e468c3a8f21d1e7882501340d1e52d16bd910 - Dec 24, 2016
https://github.com/coala/coala-bears/commit/0a99c06853f57474cd4be44fb3164bbf70cca922 - Mar 22, 2016

Invoking a command unnecessarily makes for bad reading, and the second command could also cause side effects which are undesirable and hard to understand because the exit code is from the first command.